tests/tests/operation_tests.cpp

```diff
@@ -2336,6 +2336,347 @@ BOOST_AUTO_TEST_CASE( vesting_balance_withdraw_test )
    // TODO:  Test with non-core asset and Bob account
 } FC_LOG_AND_RETHROW() }
 
+BOOST_AUTO_TEST_CASE( trade_amount_equals_zero_settle )
+{
+   try {
+      // get around Graphene issue #615 feed expiration bug
+      generate_blocks(HARDFORK_615_TIME);
+      generate_block();
+      set_expiration( db, trx );
+
+      ACTORS((paul)(michael)(rachel)(alice));
+
+      // create assets
+      const auto& bitusd = create_bitasset("USDBIT", paul_id);
+      const auto& core   = asset_id_type()(db);
+      asset_id_type bitusd_id = bitusd.id;
+      asset_id_type core_id = core.id;
+
+      // fund accounts
+      transfer(committee_account, michael_id, asset( 100000000 ) );
+      transfer(committee_account, paul_id, asset(10000000));
+      transfer(committee_account, alice_id, asset(10000000));
+
+      // add a feed to asset
+      update_feed_producers( bitusd, {paul.id} );
+      price_feed current_feed;
+      current_feed.maintenance_collateral_ratio = 1750;
+      current_feed.maximum_short_squeeze_ratio = 1100;
+      current_feed.settlement_price = bitusd.amount( 100 ) / core.amount(5);
+      publish_feed( bitusd, paul, current_feed );
+
+      // paul gets some bitusd
+      auto call_paul = *borrow( paul, bitusd.amount(1000), asset(100));
```

Comment:
Apparently we need a reference here.
If I'm correct, the `auto` here is wrong, it will copy the object.
Fortunately we didn't change the data on the right in this test case before checking with the object on the left.

Please also check other similar code in this PR.